libsolidity/parsing/Token.cpp

```diff
@@ -101,14 +109,14 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-unsigned Token::extractM(string const& _literal)
+unsigned Token::extractUnsigned(string::const_iterator _begin, string::const_iterator _end)
 {
 	try
 	{
-		unsigned short m = stoi(_literal);
+		unsigned short m = boost::lexical_cast<unsigned short>(boost::make_iterator_range(_begin, _end));
 		return m;
 	}
-	catch(out_of_range& e)
+	catch(const boost::bad_lexical_cast &)
 	{
 		return 0;
```

Comment:
Actually `fixed0x7` is a valid type, so I think this will parse `fixed99999999999999999999999999999999999999x7` as `fixed0x7` and thus we have to find another way to report an error. Perhaps make the return type `int` and return `-1` on failure (and of course check that).